### PR TITLE
feat: implement tests for check_ok and check_ping 

### PR DESCRIPTION
**Pull request by Valeri0p**
Still incomplete, there's probably some syntax error, not sure if mocking the various calls is needed.

### CODE REVIEW

File: tests/main_test.py [hunk 1]
```diff
@@ -21,8 +21,33 @@
...
+        'is_async': False
+    },
+    {
+        'func': main.check_ping,
+        'expected_res': [False],
+        'arg': ('garbage',),
+        'is_async': False
     }
 ]
+#last 4 tests are incomplete; might want to mock some calls
```
Comment (domenicoblanco) [inline tests/main_test.py]: - [x] Add test to a page that doesn't exist in order to obtain a 404
- [ ] Is really helpful test without a valid URL?
- [ ] `check_ping` should mock `call` in order to obtain some useful results

File: tests/main_test.py [hunk 2]
```diff
@@ -45,5 +70,6 @@ async def test_init(mocker: MockerFixture) -> None:
     mocker.patch.object(main, "__name__", "__main__")
     mocker.patch.object(main, 'main', return_value=None)
 
+    #this is broken
     assert await main.init() == None
```
Comment (domenicoblanco) [inline tests/main_test.py]: As #16 with the huge refactor, we moved from an async init to a synchronous one, it should be enough to remove the `await` and `async`

**Review by domenicoblanco — CHANGES_REQUESTED** [review]

File: tests/main_test.py [hunk 3]
```diff
@@ -70,6 +70,5 @@ async def test_init(mocker: MockerFixture) -> None:
     mocker.patch.object(main, "__name__", "__main__")
     mocker.patch.object(main, 'main', return_value=None)
 
-    #this is broken
-    assert await main.init() == None
+    main.init() == None
```
Comment (domenicoblanco) [inline tests/main_test.py]: The assert is still mandatory 😅

Comment (Valeri0p) [reply tests/main_test.py]: whoops

File: tests/main_test.py (hunk 1 above)
Comment (Valeri0p) [reply tests/main_test.py]: 1) done
2) not really, the urls list is managed by us
3) not needed

File: tests/main_test.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+from asyncio import subprocess
```
Comment (domenicoblanco) [inline tests/main_test.py]: Is it an unused import?

Comment (Valeri0p) [reply tests/main_test.py]: probably; thanks Vs code

**Review by domenicoblanco — APPROVED** [review]

**Pull request merged**